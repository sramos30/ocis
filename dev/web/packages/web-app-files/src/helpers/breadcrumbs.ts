import { bus } from 'web-pkg/src/instance'

export interface BreadcrumbItem {
  text: string
  to?: string
  allowContextActions?: boolean
  onClick?: () => void
}

export const breadcrumbsFromPath = (
  currentPath: string,
  resourcePath: string
): BreadcrumbItem[] => {
  const pathSplit = (p = '') => p.split('/').filter(Boolean)
  const current = pathSplit(currentPath)
  const resource = pathSplit(resourcePath)

  return resource.map(
    (text, i) =>
      ({
        allowContextActions: true,
        text,
        to: '/' + [...current].splice(0, current.length - resource.length + i + 1).join('/')
      } as BreadcrumbItem)
  )
}

export const concatBreadcrumbs = (...items: BreadcrumbItem[]): BreadcrumbItem[] => {
  const last = items.pop()

  return [
    ...items,
    {
      allowContextActions: last.allowContextActions,
      text: last.text,
      onClick: () => bus.publish('app.files.list.load')
    }
  ]
}
