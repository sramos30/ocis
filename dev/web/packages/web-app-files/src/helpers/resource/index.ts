export * from './asset'
export * from './filter'
export * from './privatePreviewBlob'
export * from './publicPreviewUrl'
export * from './resource'
export * from './sameResource'
