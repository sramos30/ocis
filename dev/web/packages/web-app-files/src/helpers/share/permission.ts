// dummy to trick gettext string extraction into recognizing strings
const $gettext = (str) => {
  return str
}

export class SharePermission {
  private readonly _key: string
  private readonly _bit: number
  private readonly _label: string

  constructor(key: string, bit: number, label: string) {
    this._key = key
    this._bit = bit
    this._label = label
  }

  get key(): string {
    return this._key
  }

  get bit(): number {
    return this._bit
  }

  get label(): string {
    return $gettext(this._label)
  }

  public enabled(bitmask: number): boolean {
    return (bitmask & this.bit) > 0
  }
}

export abstract class SharePermissions {
  static readonly read = new SharePermission('read', 1, $gettext('Read'))
  static readonly update = new SharePermission('update', 2, $gettext('Update'))
  static readonly create = new SharePermission('create', 4, $gettext('Create'))
  static readonly delete = new SharePermission('delete', 8, $gettext('Delete'))
  static readonly share = new SharePermission('share', 16, $gettext('Share'))

  static permissionsToBitmask(permissions: SharePermission[]): number {
    return (permissions || []).reduce((b: number, p: SharePermission) => b | p.bit, 0)
  }

  static bitmaskToPermissions(bitmask: number): SharePermission[] {
    return [this.read, this.update, this.create, this.delete, this.share].reduce(
      (a: SharePermission[], p: SharePermission) => {
        if (p.enabled(bitmask)) {
          a.push(p)
        }
        return a
      },
      []
    )
  }
}
