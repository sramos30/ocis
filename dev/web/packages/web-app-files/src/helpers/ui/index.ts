export * as fileList from './filesList'
export * as resourceTable from './resourceTable'
