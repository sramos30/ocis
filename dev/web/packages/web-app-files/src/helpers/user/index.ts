export * from './avatarUrl'
export * from './types'
