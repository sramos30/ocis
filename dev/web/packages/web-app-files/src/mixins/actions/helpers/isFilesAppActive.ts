import { activeApp } from 'web-pkg/src/composables'

const isFilesAppActive = (activeApp: string): boolean => {
  // FIXME: we should use this constant but it somehow breaks the unit tests
  // return activeApp === FilesApp.appInfo.id
  return activeApp === 'files'
}

export default {
  computed: {
    $_isFilesAppActive(): boolean {
      return isFilesAppActive(activeApp(this.$route))
    }
  }
}
