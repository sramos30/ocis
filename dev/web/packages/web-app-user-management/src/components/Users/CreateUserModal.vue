<template>
  <oc-modal
    :title="$gettext('Create user')"
    :button-cancel-text="$gettext('Cancel')"
    :button-confirm-text="$gettext('Create')"
    :button-confirm-disabled="buttonConfirmDisabled"
    focus-trap-initial="#input-user-name"
    @cancel="$emit('cancel')"
    @confirm="$emit('confirm', user)"
  >
    <template #content>
      <oc-text-input
        id="input-user-name"
        v-model="user.onPremisesSamAccountName"
        class="oc-mb-s"
        :label="$gettext('User name') + '*'"
        :error-message="formData.userName.errorMessage"
        :fix-message-line="true"
        @input="validateUserName"
      />
      <oc-text-input
        id="input-display-name"
        v-model="user.displayName"
        class="oc-mb-s"
        :label="$gettext('First and last name') + '*'"
        :error-message="formData.displayName.errorMessage"
        :fix-message-line="true"
        @input="validateDisplayName"
      />
      <oc-text-input
        id="input-mail"
        v-model="user.mail"
        class="oc-mb-s"
        :label="$gettext('Email') + '*'"
        :error-message="formData.email.errorMessage"
        type="email"
        :fix-message-line="true"
        @change="validateEmail"
      />
      <oc-text-input
        id="input-password"
        v-model="user.passwordProfile.password"
        class="oc-mb-s"
        :label="$gettext('Password') + '*'"
        :error-message="formData.password.errorMessage"
        type="password"
        :fix-message-line="true"
        @input="validatePassword"
      />
    </template>
  </oc-modal>
</template>

<script>
import * as EmailValidator from 'email-validator'

export default {
  name: 'CreateUserModal',
  props: {
    existingUsers: {
      type: Array,
      required: false,
      default: () => {
        return []
      }
    }
  },
  data: function () {
    return {
      formData: {
        userName: {
          errorMessage: '',
          valid: false
        },
        displayName: {
          errorMessage: '',
          valid: false
        },
        email: {
          errorMessage: '',
          valid: false
        },
        password: {
          errorMessage: '',
          valid: false
        }
      },
      user: {
        onPremisesSamAccountName: '',
        displayName: '',
        mail: '',
        passwordProfile: {
          password: ''
        }
      }
    }
  },
  computed: {
    buttonConfirmDisabled() {
      return Object.keys(this.formData)
        .map((k) => !!this.formData[k].valid)
        .includes(false)
    }
  },
  methods: {
    validateUserName() {
      this.formData.userName.valid = false

      if (this.user.onPremisesSamAccountName.trim() === '') {
        this.formData.userName.errorMessage = this.$gettext('User name cannot be empty')
        return false
      }

      if (this.user.onPremisesSamAccountName.includes(' ')) {
        this.formData.userName.errorMessage = this.$gettext('User name cannot contain white spaces')
        return false
      }

      if (
        this.user.onPremisesSamAccountName.length &&
        !isNaN(this.user.onPremisesSamAccountName[0])
      ) {
        this.formData.userName.errorMessage = this.$gettext('User name cannot start with a number')
        return false
      }

      if (
        this.existingUsers.find(
          (existingUser) =>
            existingUser.onPremisesSamAccountName === this.user.onPremisesSamAccountName
        )
      ) {
        this.formData.userName.errorMessage = this.$gettextInterpolate(
          this.$gettext('User "%{userName}" already exists'),
          { userName: this.user.onPremisesSamAccountName }
        )
        return false
      }

      this.formData.userName.errorMessage = ''
      this.formData.userName.valid = true
      return true
    },

    validateDisplayName() {
      this.formData.displayName.valid = false

      if (this.user.displayName.trim() === '') {
        this.formData.displayName.errorMessage = this.$gettext('Display name cannot be empty')
        return false
      }

      this.formData.displayName.errorMessage = ''
      this.formData.displayName.valid = true
      return true
    },

    validateEmail() {
      this.formData.email.valid = false

      if (!EmailValidator.validate(this.user.mail)) {
        this.formData.email.errorMessage = this.$gettext('Please enter a valid email')
        return false
      }

      this.formData.email.errorMessage = ''
      this.formData.email.valid = true
      return true
    },
    validatePassword() {
      this.formData.password.valid = false

      if (this.user.passwordProfile.password.trim() === '') {
        this.formData.password.errorMessage = this.$gettext('Password cannot be empty')
        return false
      }

      this.formData.password.errorMessage = ''
      this.formData.password.valid = true
      return true
    }
  }
}
</script>
