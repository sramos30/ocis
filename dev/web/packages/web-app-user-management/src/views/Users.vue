<template>
  <div>
    <create-user-modal
      v-if="createUserModalOpen"
      :existing-users="users"
      @cancel="toggleCreateUserModal"
      @confirm="createUser"
    />
    <delete-user-modal
      v-if="deleteUserModalOpen"
      :users="selectedUsers"
      @cancel="toggleDeleteUserModal"
      @confirm="deleteUsers"
    />
    <main class="oc-flex oc-flex-column oc-height-1-1 oc-p-m">
      <app-loading-spinner v-if="loadResourcesTask.isRunning" />
      <template v-else>
        <div class="oc-app-bar">
          <div class="oc-flex oc-flex-between">
            <oc-breadcrumb class="oc-flex oc-flex-middle" :items="breadcrumbs" />
          </div>
          <div class="oc-flex-1 oc-flex oc-flex-start oc-mt-m">
            <div v-if="selectedUsers.length" class="oc-flex oc-flex-middle">
              <span v-text="selectedUsersText" />
              <oc-button
                id="files-clear-selection"
                v-oc-tooltip="$gettext('Clear selection')"
                :aria-label="$gettext('Clear selection')"
                class="oc-ml-m"
                appearance="outline"
                @click="unselectAllUsers"
              >
                <oc-icon name="close" />
              </oc-button>
              <oc-button appearance="outline" class="oc-ml-m" @click="toggleDeleteUserModal">
                <oc-icon name="delete-bin" />
                <translate>Delete</translate>
              </oc-button>
            </div>
            <div v-else>
              <oc-button variation="primary" appearance="filled" @click="toggleCreateUserModal">
                <oc-icon name="add" />
                <translate>New user</translate>
              </oc-button>
            </div>
          </div>
        </div>
        <no-content-message
          v-if="!users.length"
          id="user-management-users-empty"
          class="files-empty"
          icon="user"
        >
          <template #message>
            <span v-translate>No users in here</span>
          </template>
        </no-content-message>
        <div v-else>
          <UsersList
            :users="users"
            :roles="roles"
            :user-assignments="userAssignments"
            :selected-users="selectedUsers"
            class="oc-mt-m"
            @toggleSelectUser="toggleSelectUser"
            @toggleSelectAllUsers="toggleSelectAllUsers"
          />
        </div>
      </template>
    </main>
  </div>
</template>

<script>
import UsersList from '../components/Users/UsersList.vue'
import CreateUserModal from '../components/Users/CreateUserModal.vue'
import DeleteUserModal from '../components/Users/DeleteUserModal.vue'
import AppLoadingSpinner from 'web-pkg/src/components/AppLoadingSpinner.vue'
import NoContentMessage from 'web-pkg/src/components/NoContentMessage.vue'
import { useStore } from 'web-pkg/src/composables'
import { ref } from '@vue/composition-api'
import { clientService } from 'web-pkg/src/services'
import { useTask } from 'vue-concurrency'
import { bus } from 'web-pkg/src/instance'
import { mapActions } from 'vuex'
import axios from 'axios'

export default {
  components: { UsersList, AppLoadingSpinner, NoContentMessage, CreateUserModal, DeleteUserModal },
  setup() {
    const store = useStore()
    const users = ref([])
    const roles = ref([])
    const userAssignments = ref([])
    const graphClient = clientService.graphAuthenticated(
      store.getters.configuration.server,
      store.getters.getToken
    )

    const loadRolesTask = useTask(function* (signal, ref) {
      const rolesResponse = yield axios.post(
        '/api/v0/settings/roles-list',
        {},
        {
          headers: {
            authorization: store.getters.getToken
          }
        }
      )
      roles.value = rolesResponse.data.bundles
    })

    const loadUserAssignmentTask = useTask(function* (signal, ref) {
      const userAssignmentResponse = yield axios.post(
        '/api/v0/settings/assignments-list',
        {
          account_uuid: ref.user?.id
        },
        {
          headers: {
            authorization: store.getters.getToken
          }
        }
      )
      userAssignments.value.push(userAssignmentResponse.data?.assignments)
    })

    const loadResourcesTask = useTask(function* (signal, ref) {
      const usersResponse = yield graphClient.users.listUsers('displayName')
      users.value = usersResponse.data.value || []

      yield loadRolesTask.perform()

      for (const user of users.value) {
        yield loadUserAssignmentTask.perform({ user })
      }
    })

    return {
      users,
      roles,
      userAssignments,
      loadResourcesTask,
      graphClient
    }
  },
  data: function () {
    return {
      selectedUsers: [],
      createUserModalOpen: false,
      deleteUserModalOpen: false
    }
  },
  computed: {
    selectedUsersText() {
      const translated = this.$gettext('%{ userCount } selected')

      return this.$gettextInterpolate(translated, { userCount: this.selectedUsers.length })
    },
    breadcrumbs() {
      return [
        { text: this.$gettext('User management'), to: { path: '/user-management' } },
        {
          text: this.$gettext('Users'),
          onClick: () => bus.publish('app.user-management.list.load')
        }
      ]
    },

    allUsersSelected() {
      return this.users.length === this.selectedUsers.length
    }
  },

  async mounted() {
    await this.loadResourcesTask.perform(this)

    const loadResourcesEventToken = bus.subscribe('app.user-management.list.load', () => {
      this.loadResourcesTask.perform(this)
    })

    this.$on('beforeDestroy', () => {
      bus.unsubscribe('app.user-management.list.load', loadResourcesEventToken)
    })
  },

  methods: {
    ...mapActions(['showMessage']),
    toggleSelectAllUsers() {
      if (this.allUsersSelected) {
        return (this.selectedUsers = [])
      }
      this.selectedUsers = [...this.users]
    },

    toggleSelectUser(toggledUser) {
      const isUserSelected = this.selectedUsers.find((user) => user.id === toggledUser.id)

      if (!isUserSelected) {
        return this.selectedUsers.push(toggledUser)
      }

      this.selectedUsers = this.selectedUsers.filter((user) => user.id !== toggledUser.id)
    },

    unselectAllUsers() {
      this.selectedUsers = []
    },

    toggleCreateUserModal() {
      this.createUserModalOpen = !this.createUserModalOpen
    },

    toggleDeleteUserModal() {
      this.deleteUserModalOpen = !this.deleteUserModalOpen
    },
    async deleteUsers(users) {
      const promises = users.map((user) => this.graphClient.users.deleteUser(user.id))

      try {
        await Promise.all(promises)
        this.showMessage({
          title: this.$gettextInterpolate(
            this.$ngettext(
              'User "%{user}" was deleted successfully',
              '%{userCount} users were deleted successfully',
              this.selectedUsers.length
            ),
            {
              userCount: this.selectedUsers.length,
              user: this.selectedUsers[0].onPremisesSamAccountName
            },
            true
          )
        })
        this.users = this.users.filter((user) => {
          return !users.find((deletedUser) => user.id === deletedUser.id)
        })
        this.selectedUsers = []
        this.toggleDeleteUserModal()
      } catch (error) {
        console.error(error)
        this.showMessage({
          title: this.$gettextInterpolate(
            this.$ngettext(
              'Failed to delete user "%{user}"',
              'Failed to delete %{userCount} users',
              this.selectedUsers.length
            ),
            {
              userCount: users.length,
              user: users.onPremisesSamAccountName
            },
            true
          ),
          status: 'danger'
        })
      }
    },

    async createUser(user) {
      try {
        const response = await this.graphClient.users.createUser(user)
        this.toggleCreateUserModal()
        this.showMessage({
          title: this.$gettext('User was created successfully')
        })
        this.users.push(response?.data)
      } catch (error) {
        console.error(error)
        this.showMessage({
          title: this.$gettext('Failed to create user'),
          status: 'danger'
        })
      }
    }
  }
}
</script>
