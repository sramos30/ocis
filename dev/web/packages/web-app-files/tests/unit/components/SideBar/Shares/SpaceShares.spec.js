import SpaceShares from '@files/src/components/SideBar/Shares/SpaceShares.vue'
import { createLocalVue, mount, shallowMount } from '@vue/test-utils'
import GetTextPlugin from 'vue-gettext'
import Vuex from 'vuex'
import DesignSystem from 'owncloud-design-system'
import Users from '@/__fixtures__/users'
import { ShareTypes, spaceRoleManager, spaceRoleViewer } from '../../../../../src/helpers/share'
import VueCompositionAPI from '@vue/composition-api/dist/vue-composition-api'

const localVue = createLocalVue()
localVue.use(DesignSystem)
localVue.use(Vuex)
localVue.use(VueCompositionAPI)
localVue.use(GetTextPlugin, {
  translations: 'does-not-matter.json',
  silent: true
})

const user = Users.alice
const spaceMock = {
  type: 'space',
  name: ' space',
  id: '1',
  mdate: 'Wed, 21 Oct 2015 07:28:00 GMT',
  spaceQuota: {
    used: 100,
    total: 1000
  }
}

const outgoingShares = [
  {
    id: '1',
    shareType: ShareTypes.space.value,
    collaborator: {
      name: user.id,
      displayName: user.displayname
    },
    role: {
      name: spaceRoleManager.name
    }
  },
  {
    id: '2',
    shareType: ShareTypes.space.value,
    collaborator: {
      onPremisesSamAccountName: 'Einstein',
      displayName: 'einstein'
    },
    role: {
      name: spaceRoleViewer.name
    }
  },
  {
    id: '3',
    shareType: ShareTypes.space.value,
    collaborator: {
      onPremisesSamAccountName: 'Marie',
      displayName: 'marie'
    },
    role: {
      name: spaceRoleManager.name
    }
  }
]

describe('SpaceShares', () => {
  it('renders loading spinner while loading', () => {
    const wrapper = getShallowMountedWrapper(
      {
        user
      },
      true
    )
    expect(wrapper).toMatchSnapshot()
  })
  describe('if currentUser can share', () => {
    it('initially renders add people dialog', () => {
      const wrapper = getShallowMountedWrapper({
        user,
        outgoingCollaborators: [outgoingShares[0]]
      })
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('if currentUser can not share', () => {
    it('other shares are listed, but creating/editing shares is not possible', () => {
      const wrapper = getShallowMountedWrapper({
        user,
        outgoingCollaborators: [outgoingShares[1]]
      })
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('if currentUser is manager', () => {
    it('allows role edit of the current user if another user is manager', () => {
      const wrapper = getShallowMountedWrapper({
        user,
        outgoingCollaborators: outgoingShares
      })
      expect(wrapper).toMatchSnapshot()
    })
    it('does not allow role edit of the current user if they are the only manager', () => {
      const wrapper = getShallowMountedWrapper({
        user,
        outgoingCollaborators: [outgoingShares[0], outgoingShares[1]]
      })
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('if currentUser can delete', () => {
    it('reacts on delete events by collaborator list items', async () => {
      const wrapper = getMountedWrapper({
        user,
        outgoingCollaborators: outgoingShares
      })

      const spyOnCollaboratorDelete = jest.spyOn(wrapper.vm, 'deleteShare')
      wrapper
        .find(`div[data-testid="collaborator-user-item-${outgoingShares[0].collaborator.name}"]`)
        .vm.$emit('onDelete')
      await wrapper.vm.$nextTick()
      expect(spyOnCollaboratorDelete).toHaveBeenCalledTimes(1)
    })
  })
})

const storeOptions = (data, isInLoadingState) => {
  const { user, outgoingCollaborators = [] } = data

  return {
    state: {
      user
    },
    modules: {
      Files: {
        namespaced: true,
        getters: {
          highlightedFile: () => spaceMock,
          currentFileOutgoingCollaborators: () => outgoingCollaborators,
          currentFileOutgoingSharesLoading: () => isInLoadingState,
          sharesTreeLoading: () => false
        },
        actions: {
          loadCurrentFileOutgoingShares: jest.fn(),
          deleteShare: jest.fn()
        },
        mutations: {
          SET_HIGHLIGHTED_FILE(state, file) {
            state.highlightedFile = spaceMock
          }
        }
      }
    },
    getters: {
      user: () => user,
      capabilities: () => {
        return {
          files_sharing: {
            user: {
              expire_date: {
                enabled: true,
                days: 10
              }
            },
            group: {
              expire_date: {
                enabled: true,
                days: 10
              }
            }
          }
        }
      }
    }
  }
}

function getShallowMountedWrapper(data, loading = false) {
  return shallowMount(getComponent(loading), {
    localVue,
    store: createStore(data, loading),
    stubs: {
      'oc-button': true,
      'oc-icon': true,
      'oc-spinner': true
    },
    provide: {
      displayedItem: {
        value: spaceMock
      }
    }
  })
}

function getMountedWrapper(data, loading = false) {
  return mount(getComponent(loading), {
    localVue,
    store: createStore(data, loading),
    stubs: {
      'oc-button': true,
      'oc-icon': true,
      'oc-spinner': true,
      'avatar-image': true,
      'role-dropdown': true,
      'edit-dropdown': true
    },
    mocks: {
      $router: {
        go: jest.fn(),
        push: jest.fn(),
        currentRoute: {
          name: 'some-route',
          query: { page: 1 }
        },
        resolve: (r) => ({ href: r.name })
      }
    },
    provide: {
      displayedItem: {
        value: spaceMock
      }
    }
  })
}

function createStore(data, loading) {
  return new Vuex.Store(storeOptions(data, loading))
}

function getComponent(loading) {
  return {
    ...SpaceShares,
    setup: () => ({
      graphClient: {},
      loadSharesTask: {
        isRunning: loading,
        perform: jest.fn()
      }
    })
  }
}
