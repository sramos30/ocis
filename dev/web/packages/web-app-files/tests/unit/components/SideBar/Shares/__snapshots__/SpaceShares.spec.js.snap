// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpaceShares if currentUser can not share other shares are listed, but creating/editing shares is not possible 1`] = `
<div id="oc-files-sharing-sidebar" class="oc-position-relative">
  <!---->
  <ul id="files-collaborators-list" aria-label="Share receivers" class="oc-list oc-list-divider oc-overflow-hidden oc-m-rm">
    <li>
      <collaborator-list-item-stub share="[object Object]"></collaborator-list-item-stub>
    </li>
  </ul>
</div>
`;

exports[`SpaceShares if currentUser can share initially renders add people dialog 1`] = `
<div id="oc-files-sharing-sidebar" class="oc-position-relative">
  <invite-collaborator-form-stub class="oc-my-s"></invite-collaborator-form-stub>
  <ul id="files-collaborators-list" aria-label="Share receivers" class="oc-list oc-list-divider oc-overflow-hidden oc-m-rm">
    <li>
      <collaborator-list-item-stub share="[object Object]"></collaborator-list-item-stub>
    </li>
  </ul>
</div>
`;

exports[`SpaceShares if currentUser is manager allows role edit of the current user if another user is manager 1`] = `
<div id="oc-files-sharing-sidebar" class="oc-position-relative">
  <invite-collaborator-form-stub class="oc-my-s"></invite-collaborator-form-stub>
  <ul id="files-collaborators-list" aria-label="Share receivers" class="oc-list oc-list-divider oc-overflow-hidden oc-m-rm">
    <li>
      <collaborator-list-item-stub share="[object Object]" modifiable="true"></collaborator-list-item-stub>
    </li>
    <li>
      <collaborator-list-item-stub share="[object Object]" modifiable="true"></collaborator-list-item-stub>
    </li>
    <li>
      <collaborator-list-item-stub share="[object Object]" modifiable="true"></collaborator-list-item-stub>
    </li>
  </ul>
</div>
`;

exports[`SpaceShares if currentUser is manager does not allow role edit of the current user if they are the only manager 1`] = `
<div id="oc-files-sharing-sidebar" class="oc-position-relative">
  <invite-collaborator-form-stub class="oc-my-s"></invite-collaborator-form-stub>
  <ul id="files-collaborators-list" aria-label="Share receivers" class="oc-list oc-list-divider oc-overflow-hidden oc-m-rm">
    <li>
      <collaborator-list-item-stub share="[object Object]"></collaborator-list-item-stub>
    </li>
    <li>
      <collaborator-list-item-stub share="[object Object]" modifiable="true"></collaborator-list-item-stub>
    </li>
  </ul>
</div>
`;

exports[`SpaceShares renders loading spinner while loading 1`] = `
<div id="oc-files-sharing-sidebar" class="oc-position-relative">
  <oc-loader-stub arialabel="Loading members"></oc-loader-stub>
</div>
`;
